### PR TITLE
Needed scope

### PR DESCRIPTION
**Pull request by gpenverne**

### CODE REVIEW

File: SDK.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        $data = curl_exec($ch);
+        if (curl_errno($ch)) {
+            throw new \Exception('Error:'.curl_error($ch));
+        }
+        curl_close($ch);
+
+        return json_decode($data);
+    }
+
+    private function getUrl($url)
```
Comment (ndavoust) [inline SDK.php]: ENDPOINT becomes $url here... inconsistent :)

File: SDK.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    private function query($endpoint, $parameters = [])
+    {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $this->getUrl($endpoint));
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+        if ($parameters && !empty($parameters)) {
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $parameters);
+        }
+        $headers = ['AUTHORIZATION: '.sprintf('Bearer %s', $this->getAccessToken())];
```
Comment (ndavoust) [inline SDK.php]: sprintf('AUTHORIZATION : Bearer %s', $this->getAccessToken()) ?

File: SDK.php [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        $this->jsonToken = $stdClass;
+    }
+
+    public function getAccessToken($code = null)
+    {
+        if (null !== $this->jsonToken) {
+            return $this->jsonToken->access_token;
+        }
+
+        if (null === $code && isset($_GET['code'])) {
```
Comment (ndavoust) [inline SDK.php]: force getAccessToken($code) and skip this if condition?

File: SDK.php [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+    public function __construct($clientId, $clientSecret, $redirectUri, $scope = [self::SCOPE_PROFILE])
+    {
+        $this->clientId = $clientId;
+        $this->clientSecret = $clientSecret;
+        $this->redirectUri = $redirectUri;
+        $this->scope = $scope;
+
+        if (isset($_GET['code'])) {
+            $this->getAccessToken();
```
Comment (ndavoust) [inline SDK.php]: $this->getAccessToken($_GET['code'])

File: SDK.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+redirectUri.php
+```php
+
+// Include the SDK file
+include "SDK.php";
+
+// Prepare some configuration values
+$clientId = 'my-client-id';
+$clientSecret = 'my-client-secret';
+$redirectUri = 'http://my-domain.com/redirectUri.php';
```
Comment (ndavoust) [inline SDK.md]: https://myapp.io/connect/gdc to be consistent?

File: README.md [hunk 6]
```diff
@@ -20,47 +20,11 @@ At the end of the application's registration, you get your secret application ke
 In your application configuration, you have to use these informations to use SSO :
 
 - Authorization URL : `https://gensdeconfiance.fr/oauth/v2/auth` (used to login)
-- Access token URL : `https://gensdeconfiance.fr/oauth/v2/token`
+- Access token URL : `https://gensdeconfiance.fr/oauth/v2/token`
```
Comment (kbsali) [inline README.md]: NOP

File: SDK.php (hunk 2 above)
Comment (kbsali) [reply SDK.php]: +1

File: SDK.php (hunk 3 above)
Comment (gpenverne) [reply SDK.php]: No. If we want to retrieve the current access token, we don't care about the $code

File: SDK.php (hunk 4 above)
Comment (gpenverne) [reply SDK.php]: Without the isset test , we will have a warning

File: SDK.md (hunk 5 above)
Comment (gpenverne) [reply SDK.md]: Ok for domain name, but keep the redirectUri.php to let know that the file will be the endpoint

File: SDK.php (hunk 4 above)
Comment (ndavoust) [reply SDK.php]: I would have let the isset test, just passing $this->getAccessToken($_GET['code']) ;)

**Pull request merged**